File: skills/unsloth-lora/scripts/unsloth-lora_tool.js
/**
 * Helper to format LoRA parameters for documentation or configuration files.
 */
function getLoRAConfig(rank, alpha, useRslora = false) {
    return {
        r: rank,
        lora_alpha: alpha,
        use_rslora: useRslora,
        lora_dropout: 0, // Recommended by Unsloth for speed
        target_modules: ["q_proj", "k_proj", "v_proj", "o_proj", "gate_proj", "up_proj", "down_proj"]
    };
}

const args = process.argv.slice(2);
const r = parseInt(args[0]) || 16;
const a = parseInt(args[1]) || 16;
const rs = args[2] === 'true';

console.log(JSON.stringify(getLoRAConfig(r, a, rs), null, 2));

File: skills/unsloth-lora/scripts/unsloth-lora_tool.py
import argparse

def configure_lora(rank, alpha, use_rslora=False, dropout=0.0):
    """
    Generates the configuration for Unsloth get_peft_model.
    """
    config = {
        "r": rank,
        "target_modules": [
            "q_proj", "k_proj", "v_proj", "o_proj",
            "gate_proj", "up_proj", "down_proj",
        ],
        "lora_alpha": alpha,
        "lora_dropout": dropout,
        "bias": "none",
        "use_gradient_checkpointing": "unsloth",
        "random_state": 3407,
        "use_rslora": use_rslora,
    }
    return config

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Unsloth LoRA Configurator")
    parser.add_argument("--rank", type=int, default=16)
    parser.add_argument("--alpha", type=int, default=16)
    parser.add_argument("--rslora", action="store_true")
    args = parser.parse_args()
    
    print(f"Generated Config: {configure_lora(args.rank, args.alpha, args.rslora)}")